### ISSUE TITLE
Bug: [no-extra-parens] incorrect precedence for await vs assertion

### ISSUE THREAD

**Issue by bradzacher** [typescript-eslint/typescript-eslint#5124]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.2&sourceType=module&code=IYZwngdgxgBAZgV2gFwJYHsLwBTIBaoQDmAXDMBGAJQwDeAUDOQO7CrIy4HHkgwAKAJ3QBbVCACmAHgBu6VABMAfFQDc9AL70gA&eslintrc=N4KAkARApgHgLlAdgEwM4QFwAIDa5IAOANgK4DmAlohgAJwCeBUqAxgE4UFwC0zRVcAPR8B3NlBYB7ALbSkyKMggAafBGLkqtBk1YcuvVP0RCAhkSIQQWLAF1VNiGxJFmmLKBuO6jZu048IiaCiJLc0hSoJKiK3ARsMpFu2HheXtBsCWwq1mkeuXlYECwAFhIA1qgAapIUyABKUHAkbIjuAGbmMQVeAL499j0QPrr+BkFCobzwbKZxpuKI6Ck9jlCZktkD1v29QA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkAhqqrAO4BSZlemArugDTgaMABbpoAa3YUwAMwYl+g2iXHJEAGWQAjAMKiJ0rrwE0oRWAFF4yPIZ6LTkACYsiGAG7pUASTxY6BMR2xkpQJLDcmNDoALJ0iMEOQiRcyNC2nPYmQubeSKhpNgCCRACeidnKqekActyMemLiHGBGSbS5+YV4ACrCyC1yqAqVZrAAqkTcCk7qsNDyFaGQ5pPT6E4ACnSYdPDofpiD8u1jeYgF0DYASgeRpEuO5gBiDKh4wpgRAObCOnQKEjeIgAZWYNmgwkeOQmREhTQ2wKc6AAHhsitBoiRjsNTpB1pNxOZmERgf88JCAGo7ZB0LQYFptUaQVF0dK4QJEBibL6ILB4Uo9Up8nEjZboEgxWBOeroYGHWAJTIhRzI6CoHYBQhEaRDMWOZBIWCYPAACS8fKO0NoeB23wO0hZJAATAAGACMAHZIMz4NLZY6JURUbZmQArEgox2YdBs0PLLQA9DjTCoR0AOnTAHofcsIngACLITAZ7N+JC5xwxv2eHQIfZEPCM+yCAC+2RgCBQGGwoLonl1Jw7yDhqG4yOkAG1liQolmAFTzheVoR+FJZ7hEGxrvALpfzyCCAC6HdR6vH6Cny3MyIA+n6ZQzDxAjyBW0A

### Repro Code

```typescript
async function f(thing: any) {
  await (thing as Promise<void>);
}
```

### ESLint Config

```javascript
module.exports = {
	"extends": [
		"plugin:@typescript-eslint/eslint-recommended",
		"plugin:@typescript-eslint/all"
  ],
  "rules": {
    "@typescript-eslint/no-misused-promises": [
      "error",
      {
        "checksVoidReturn": false
      }
    ],
    "@typescript-eslint/no-extra-parens": [
      "error"
    ]
 }
}
```

### Expected Result

No error as these are necessary parentheses in order to ensure the assertion is applied to the await argument, not the await result.

### Actual Result

```
Unnecessary parentheses around expression. 2:9 - 2:10
```

### Additional Info

Found in #5121 cc @doberkofler 

A big problem that I can see is that this rule uses our old-school hack-and-extend method for extending the lint rule.
Instead we should fork the lint rule and leverage our [relatively new precedence utilities](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/util/getOperatorPrecedence.ts) in order to correctly understand all cases.

### Versions

| package | version |
| -- | -- |
| `@typescript-eslint/eslint-plugin` | `5.27.0` |
| `@typescript-eslint/parser` | `5.27.0` |
| `TypeScript` | `4.7.2` |
| `ESLint` | `8.15.0` |
| `node` | `web` |

**Comment by armano2** [comment]
related #1587
